### PR TITLE
Config upload

### PR DESCRIPTION
**Pull request by bmcutler**

### CODE REVIEW

File: TAGradingServer/account/submit/admin-gradeable.php [hunk 1]
```diff
@@ -557,6 +560,30 @@ function getGraders($var){
...
+                           "gradeable" =>  $_POST['gradeable_id']);
+
+$config_build_fp = fopen($config_build_file,'w');
+
+if (!$config_build_fp){
+  die('failed to open file'.$config_build_file);
+}
+
+fwrite($config_build_fp, json_encode($config_build_data), JSON_PRETTY_PRINT);
+fclose($config_build_fp);
```
Comment (MasterOdin) [inline TAGradingServer/account/submit/admin-gradeable.php]:
```
if (file_put_contents($config_build_file, json_encode($config_build_data, JSON_PRETTY_PRINT)) === false) {
    die("Failed to write file {$config_build_file}");
}
```
as this'll handle fopen and all that for us.

File: bin/build_config_upload.py [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+	build_script = "/var/local/submitty/courses/" + semester + "/" + course + "/BUILD_" + course + ".sh"
+	build_output = "/var/local/submitty/courses/" + semester + "/" + course + "/build_script_output.txt"
+
+	f = open(build_output,'w')
+	subprocess.call([ build_script, gradeable ],stdout=f,stderr=f)
+
+
+# ------------------------------------------------------------------------
+def buildAll() :
+    for file in glob.glob("/var/local/submitty/to_be_built/*.json") :
```
Comment (MasterOdin) [inline bin/build_config_upload.py]: Please don't use `file` as a variable name as that's a builtin for python. `filename` perhaps?

**Comment by bmcutler** [comment]
@MasterOdin The upload config page still needs some work.  At the minimum, it should say what version was just uploaded (a copy paste line that can be inserted into the add-or-edit-gradeable page).  And it would be nice to have it refresh automatically when the build output finishes (requires reload now).  And it would be great to see file names and even file contents for previous uploads.

Do you want to add to this PR, or a later PR?

**Comment by MasterOdin** [comment]
I added a list of uploaded directories (without file lists) which will do for now and more features can come later.

**Comment by MasterOdin** [comment]
Travis is currently down with the AWS outage so merging this manually for now.

**Pull request merged**